content/blog/2023/2023-09-16-create-openfeign-http-client.md

```diff
@@ -0,0 +1,503 @@
+
+## Handling Responses
+Feign also provides a declarative approach to API integration. Instead of manually writing boilerplate code for handling request/response serialization/deserialization or error handling, Feign allows us to define these aspects declaratively using annotations. This not only reduces the amount of code we need to write but also improves readability and maintainability.
+
+## Handling Errors
+Error handling is a crucial aspect of building robust and reliable applications, especially when it comes to making remote API calls. Feign offers powerful features that can assist in effectively handling errors.
+
+**Feign gives us more control over handling unexpected responses. We can register a custom ErrorDecoder via the builder.**
+    
+```java
+final CalculatorService target = Feign.builder()
+                                      .errorDecoder(new CalculatorErrorDecoder())
+                                      .target(CalculatorService.class, HOST);
+
+```
+
+Here is an example to show error handling:
+```java
+public class CalculatorErrorDecoder implements ErrorDecoder {
+  private final ErrorDecoder defaultErrorDecoder = new Default();
+
+  @Override
+  public Exception decode(String methodKey, Response response) {
+    ExceptionMessage message = null;
+    try (InputStream bodyIs = response.body().asInputStream()) {
+      ObjectMapper mapper = new ObjectMapper();
+      message = mapper.readValue(bodyIs, ExceptionMessage.class);
+    } catch (IOException e) {
+      return new Exception(e.getMessage());
+    }
+    final String messageStr = message == null ? "" : message.getMessage();
+    switch (response.status()) {
+      case 400:
+        return new RuntimeException(messageStr.isEmpty() ? "Bad Request" : messageStr);
+      case 401:
+        return new RetryableException(response.status(),
+                                      response.reason(),
+                                      response.request().httpMethod(),
+                                      null,
+                                      response.request());
+      case 404:
+        return new RuntimeException(messageStr.isEmpty() ? "Not found" : messageStr);
+      default:
+        return defaultErrorDecoder.decode(methodKey, response);
+    }
+  }
+}
+```
+All responses with HTTP status other than `HTTP 2xx` range, for example `HTTP 400`, will trigger the ErrorDecoder's `decode` method. 
+In this overridden `decode` method, we can handle the response, wrap the failure into a custom exception or perform any additional processing.
+
+
+We can even retry the request again by throwing a `RetryableException`. 
+This will invoke the registered `Retryer`. [Retryer](#configuring-retryer) is explained in detail in the advanced techniques.
+
+You can see it in action by running `givenNegativeDivisorDivisionReturnsError` test in the example code shared on [Github](#example-code).
+
+## Advanced Techniques
+
+### Integrating Encoder/Decoder
```

Comment:
I would rename this section title to what you're trying to achieve. We don't want to "integrate an encoder or decoder", but we want to "Encode/decode Requests and Responses"